contrib/example-backend/http/server.go

```diff
@@ -52,26 +52,31 @@ func NewServer(options *options.Serve) (*Server, error) {
 	return server, nil
 }
 
+func (s *Server) AddRoute(path string, handler http.HandlerFunc, method string) {
+	s.server.Handler.(*mux.Router).HandleFunc(path, handler).Methods(method)
+}
+
 func (s *Server) Addr() net.Addr {
 	return s.listener.Addr()
 }
 
 func (s *Server) Start(ctx context.Context) error {
-	server := &http.Server{
-		Handler: s.Router,
-	}
 	go func() {
 		<-ctx.Done()
-		server.Close() // nolint:errcheck
+		s.server.Close() // nolint:errcheck
 	}()
 
 	go func() {
 		if s.options.KeyFile == "" {
-			server.Serve(s.listener) // nolint:errcheck
+			s.server.Serve(s.listener) // nolint:errcheck
 		} else {
-			server.ServeTLS(s.listener, s.options.CertFile, s.options.KeyFile) // nolint:errcheck
+			s.server.ServeTLS(s.listener, s.options.CertFile, s.options.KeyFile) // nolint:errcheck
 		}
 	}()
 
 	return nil
 }
+
+func (s *Server) Shutdown() error {
+	return s.server.Close()
```

Comment:
do you mind to remove this in favor of https://github.com/kube-bind/kube-bind/pull/131/commits/31aa349c6f3cf9d4495a9c09d4158aa8cb90bfe3#diff-fee0857912127761b01d43d3a59e2f9e5c226206cc14c38b12da75f96617a798R84-R86 so it's consistent in git history?

Reply:
am not convinced of turning our server into an `http.Server`. We should abstract, not inherit.